hydra-node/src/Hydra/HeadLogic.hs

```diff
@@ -205,9 +204,9 @@ update Environment{party, snapshotStrategy} ledger (HeadState parameters st) ev
   (CollectingState remainingParties _, ClientEvent (Commit utxo)) ->
     if canCommit
       then newState st [OnChainEffect (CommitTx party utxo)]
-      else panic $ "you're not allowed to commit (anymore): remainingParties : " <> show remainingParties <> ", partiyIndex:  " <> show party
+      else error $ "you're not allowed to commit (anymore): remainingParties : " <> show remainingParties <> ", partiyIndex:  " <> show party
    where
-    canCommit = party `elem` remainingParties
+    canCommit = party `Set.member` remainingParties
```

Comment:
I'd prefer the `elem` from `Foldable` .. isn't it in our prelude?

Reply:
This is discouraged by `relude` actually (and triggers a warning / deprecation notice): 

> You can't call elem and notElem functions over Set and HashSet. These functions are forbidden for these two types due to performance reasons.